client/v3/client.go

```diff
@@ -297,16 +297,23 @@ func (c *Client) dial(creds grpccredentials.TransportCredentials, dopts ...grpc.
 		dctx, cancel = context.WithTimeout(c.ctx, c.cfg.DialTimeout)
 		defer cancel() // TODO: Is this right for cases where grpc.WithBlock() is not set on the dial options?
 	}
-
-	initialEndpoints := strings.Join(c.Endpoints(), ";")
-	target := fmt.Sprintf("%s://%p/#initially=[%s]", resolver.Schema, c, initialEndpoints)
+	// Using "scheme://endpoint/authority" format. Endpoint passed doesn't matter as we are using custom resolver.
+	target := fmt.Sprintf("%s://%p/%s", resolver.Schema, c, authority(c.endpoints[0]))
 	conn, err := grpc.DialContext(dctx, target, opts...)
 	if err != nil {
 		return nil, err
 	}
 	return conn, nil
 }
 
+func authority(endpoint string) string {
```

Comment:
I think we should mimic semantic from: 

https://github.com/grpc/grpc-go/blob/e469f0d5f5bcc1324dc3940c584e0969e2ea1f90/clientconn.go#L274-L281

but TBH I don't understand the logic: 
```
		// Use endpoint from "scheme://authority/endpoint" as the default
		// authority for ClientConn.
		cc.authority = cc.parsedTarget.Endpoint
```
I.e. why authority becomes the endpoint and not the 'authority'.

Reply:
Actually looking at this doc

https://github.com/grpc/grpc/blob/master/doc/naming.md

referenced from: 
https://github.com/grpc/grpc-go/blob/e469f0d5f5bcc1324dc3940c584e0969e2ea1f90/internal/grpcutil/target.go#L47

Seems that if dns is being used, we don't want to consider DNS server an authority: `dns:[//authority/]host[:port]`
while in case of  `dns:host[:port]` it should be host:port.

Reply:
So, what's the conclusion?

Reply:
@menghanl 

I would stick to grpc behavior for a single explicitly given endpoint without client-side resolver/load-balancer, so:
```
target := initialEndpoints[0]
```

E.g. this would e.g. translate `unix://foo` to `localhost'. 

Please check the behavior for most popular target formats and please try mimic the implementation to give the same results.

Reply:
I have added integration tests that confirm that current logic works for currently supported naming schemes. I was not able to use all naming schemes from https://github.com/grpc/grpc/blob/master/doc/naming.md as most of them are not supported by etcd. Supporting additional schemes should be tracked by separate Issue and should not be backported to v3.5

Reply:
Thank you.